libcudacxx/include/cuda/__floating_point/operators.h

```diff
@@ -0,0 +1,1250 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA___FLOATING_POINT_OPERATORS_H
+#define _CUDA___FLOATING_POINT_OPERATORS_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#if _CCCL_STD_VER >= 2017
+
+#  include <cuda/__floating_point/fp.h>
+#  include <cuda/std/__concepts/concept_macros.h>
+
+#  if _CCCL_HAS_INCLUDE(<stdfloat>)
+#    include <stdfloat>
+#  endif
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+template <class _Lhs, typename _Rhs>
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI constexpr auto __fp_make_common_type()
+{
+  constexpr auto __rank_order = __fp_make_conv_rank_order<_Lhs, _Rhs>();
+
+  if constexpr (__rank_order == __fp_conv_rank_order::__equal)
+  {
+    // Extended floating point types have higher subrank, prefer cuda extended types over std extended types
+    // Fixme: potentially will not work correctly for long double
+    // auto val = 1.0f64 + 1.0l; // val will be of type long double, is this right?
+    if constexpr (__is_standard_floating_point_v<_Lhs> || __is_std_extended_floating_point_v<_Lhs>)
```

Comment:
It is true that the type of `1.0f64 + 1.0L` is always `long double` in C++.  See https://cplusplus.github.io/CWG/issues/2836.html

If `__fp_make_conv_rank_order` is fixed to handle `long double` correctly, then `long double` won't be a problem here.

If you want to prefer cuda extended type over standard extended types and and standard types, and prefer standard extended types over standard types, then the correct `if` expression is
```c++
if constexpr ((__is_standard_floating_point_v<_Lhs> || __is_std_extended_floating_point_v<_Lhs>) &&
              !__is_standard_floating_point_v<_Rhs>) {
```